io.github.shmmodbus.yml

```diff
@@ -0,0 +1,268 @@
+id: io.github.shmmodbus
+runtime: org.kde.Platform
+runtime-version: '6.6'
+sdk: org.kde.Sdk
+command: shm-modbus
+finish-args:
+  - --device=shm
+  - --share=network
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+modules:
+  - name: cxxopts
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCXXOPTS_BUILD_EXAMPLES=OFF
+      - -DCXXOPTS_BUILD_TESTS=OFF
+    sources:
+      - type: git
+        tag: v3.2.0
+        url: https://github.com/jarro2783/cxxopts.git
+
+  - name: cxxshm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+      - -DCLANG_TIDY=OFF
+      - -DBUILD_DOC=OFF
+      - -DENABLE_TEST=OFF
+    sources:
+      - type: git
+        tag: v2.0.2
+        url: https://github.com/NikolasK-source/cxxshm.git
+
+  - name: cxxsemaphore
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+      - -DCLANG_TIDY=OFF
+      - -DBUILD_DOC=OFF
+      - -DENABLE_TEST=OFF
+    sources:
+      - type: git
+        tag: v2.0.2
+        url: https://github.com/NikolasK-source/cxxsemaphore.git
+
+  - name: cxxitimer
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+      - -DCLANG_TIDY=OFF
+      - -DBUILD_DOC=OFF
+      - -DENABLE_TEST=OFF
+    sources:
+      - type: git
+        tag: v2.0.2
+        url: https://github.com/NikolasK-source/cxxitimer.git
+
+  - name: nlohmann_json
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DJSON_BuildTests=OFF
+    sources:
+      - type: git
+        tag: v3.11.3
+        url: https://github.com/nlohmann/json.git
+
+  - name: libmodbus
+    buildsystem: autotools
+    sources:
+      - type: git
+        tag: v3.1.10
+        url: https://github.com/stephane/libmodbus.git
+
+  - name: python3-PySide6
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+            --prefix=${FLATPAK_DEST} "PySide6>=6.6.2" --no-build-isolation
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/27/48/71939e07e19824d017bd2209ddc6b42e1a767268e21e43d5ed2dbd2d4df9/PySide6-6.6.3-cp38-abi3-manylinux_2_28_x86_64.whl
```

Comment:
you should add only-arches  **per source** so that they are not needessly downloaded.

Reply:
I doubt this will work as `cp38` is for Python 3.8 which is NOT the version in this runtime.

Reply:
cp38 indicates the minimum required python version, so it works just fine.